### ISSUE TITLE
Auto-updater on Mac: Error: sha512 checksum mismatch

### ISSUE THREAD

**Issue by vkolova** [electron-userland/electron-builder#3952]
* **Version**:  20.41.0

* **version of electron-updater**: 4.0.6

* **Target**: macOS

Hello! I'm getting checksum mismatch after update is downloaded. The strangest thing is that the checksum it shows does not exist anywhere (it's not the checksum shown in the latest-mac.yml for it, and it's not the one in the latest version of the app).

**Comment by vkolova** [comment]
**Update:**
I tried to manually check the checksums... and could not verify the expected one?

I build a new version of the application and tried to verify it the following ways:

```bash
openssl dgst -sha512 my-app.dmg
```

```bash
sha512sum my-app.dmg
```

```bash
shasum -a 512 my-app.dmg
```

None of which gave me the result, listed in the latest-mac.yml file.

I also tried verifying using [Checksum Validator](https://github.com/alexanderwe/checksum-validator) which also resulted in a checksum mismatch.

**Comment by JohnTendik** [comment]
For checking sha512 for you app, use the node script located in this thread:
https://github.com/electron-userland/electron-builder/issues/3913

Running that in node gave us the correct sha512 value...


As for the mismatch, I want to just put my 2cents here for anyone who stumbles upon this in the future. What our issue was that the signing process was happening asynchronously. So from time to time, the signing process would take longer than expected at which point the yaml file was generated for us hence the mismatched sha512 values. So in our case, it was a race condition against build process and sign process.. We ended up using execSync in our custom sign file and now the process hangs until the childprocess(sign process) is completely finished.. This seemed to have solved our mismatched issue. Hope this helps someone.

**Comment by nuryagdym** [comment]
In my case with `electron-updater v4.3.5` was not with the library. I was testing it library which does started giving this checksum error after upgrade. I did several builds of app and uploaded thought file must be corrupted on upload.
The problems was that each time I uploaded build files with exact the same name, and my server was caching it. As a result, I did not get the latest version of the file.
